File: tests-docker/Dockerfile.php
FROM php:latest

WORKDIR /tests
COPY tests/env_php_tests .

RUN mkdir lib && cd lib && \
    curl -sL 'https://github.com/smarty-php/smarty/archive/v3.1.29.tar.gz' | tar xzf - && \
    curl -sL 'https://github.com/twigphp/Twig/archive/v1.24.1.tar.gz' | tar xzf - && \
    curl -sL 'https://github.com/twigphp/Twig/archive/v1.19.0.tar.gz' | tar xzf -

EXPOSE 15002

CMD ["php", "-S", "0.0.0.0:15002"]

File: tests-docker/Dockerfile.java
FROM gradle:latest

WORKDIR /tests
COPY tests/env_java_tests/spark-app .

# install dependencies
USER root
RUN sed -ie 's/id "com\.github\.johnrengelman\.shadow".*//' build.gradle && \
    gradle classes

EXPOSE 15003

CMD ["gradle", "run"]
